pages/app/add-testimonial.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  Input,
+  Textarea,
+  Button,
+  Heading,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+} from '@chakra-ui/react';
+import { ChangeEvent, FormEvent, useState } from 'react';
+
+import Section from '@packages/components/Section';
+import { useI18n } from '@packages/features/i18n-context';
+import { addTestimonial } from '@packages/services/testimonials';
+import useCustomToast from '@packages/hooks/useCustomToast';
+
+interface formState {
+  name: { value: string; error: boolean };
+  gitUsername: { value: string; error: boolean };
+  testimonial: { value: string; error: boolean };
+}
+
+export default function AddTestimonialPage() {
+  const { errorToast, successToast } = useCustomToast('testimonials');
+  const { t } = useI18n('testimonials');
+  const [isLoading, setIsLoading] = useState(false);
+  const [formState, setFormState] = useState<formState>({
+    name: { value: '', error: false },
+    gitUsername: { value: '', error: false },
+    testimonial: { value: '', error: false },
+  });
+  const { name, gitUsername, testimonial } = formState;
+  const maxInputLength: number = 300;
+
+  function onChange(
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) {
+    setFormState({
+      ...formState,
+      [event.target.name]: { ...event.target, name: event.target.value },
+    });
+  }
+
+  function clearForm() {
+    setFormState({
+      name: { value: '', error: false },
+      gitUsername: { value: '', error: false },
+      testimonial: { value: '', error: false },
+    });
+  }
+
+  async function handleSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    if (checkInvalidInputs()) return errorToast('toast.inputError');
+
+    setIsLoading(true);
+    const error = await addTestimonial({
+      name: name.value,
+      testimonial: testimonial.value,
+      gitUsername: gitUsername.value,
+    });
+    setIsLoading(false);
+
+    if (error === 1) return errorToast('toast.invalidUserError');
+    if (error === 2) return errorToast('toast.serverError');
```

Comment:
Considerando meu outro commentário, sabemos que error pode assumir apenas dois valores:
`toast.invalidUserError` ou  `toast.serverError`, logo, o resto da função seria:

```suggestion
    if (error) return errorToast(error);
```

Mesmo se houvessem 50 mensagens possíveis de erro na minha função, eu não gastaria mais nenhuma linha nos clientes que a utilizam. Deixando o if aqui definir a mensagem é jogar responsabilidade da função `addTestimonial` para o componente, oque queremos evitar.